R/fortify-network.R

```diff
@@ -136,88 +116,18 @@ fortify.network <- function(
-  if (scale) {
-    nodes$x <- scale_safely(nodes$x)
-    nodes$y <- scale_safely(nodes$y)
-  }
-
-  # import vertex attributes
-  for (i in network::list.vertex.attributes(model)) {
-    nodes <- cbind(nodes, network::get.vertex.attribute(model, i), stringsAsFactors = stringsAsFactors)
-    names(nodes)[ncol(nodes)] <- i
-  }
-
-  # edge list
-  edges <- network::as.matrix.network.edgelist(model, attrname = weights)
-
-  # edge list (if there are duplicated rows)
-  if (
-    nrow(edges[, 1:2, drop = FALSE]) > nrow(unique(edges[, 1:2, drop = FALSE]))
-  ) {
-    warning("duplicated edges detected")
-  }
-
-  edges <- data.frame(nodes[edges[, 1], 1:2], nodes[edges[, 2], 1:2])
-  names(edges) <- c("x", "y", "xend", "yend")
-
-  # arrow gap (thanks to @heike and @ethen8181 for their work on this issue)
-  if (arrow.gap > 0) {
-    x.length <- edges$xend - edges$x
-    y.length <- edges$yend - edges$y
-    arrow.gap <- arrow.gap / sqrt(x.length^2 + y.length^2)
-    edges$xend <- edges$x + (1 - arrow.gap) * x.length
-    edges$yend <- edges$y + (1 - arrow.gap) * y.length
-  }
-
-  # import edge attributes
-  for (i in network::list.edge.attributes(model)) {
-    edges <- cbind(edges, network::get.edge.attribute(model, i), stringsAsFactors = stringsAsFactors)
-    names(edges)[ncol(edges)] <- i
-  }
-
-  if (nrow(edges) != 0) {
-    # merge edges and nodes data
-    edges <- merge(nodes, edges, by = c("x", "y"), all = TRUE)
-
-    # add missing columns to nodes data
-    nodes$xend <- nodes$x
-    nodes$yend <- nodes$y
-    names(nodes) <- names(edges)[1:ncol(nodes)]
-
-    # make nodes data of identical dimensions to edges data
-    for (y in names(edges)[(1 + ncol(nodes)):ncol(edges)]) {
-      nodes <- cbind(nodes, NA)
-      names(nodes)[ncol(nodes)] <- y
-    }
-
-    # panelize nodes (for temporal networks)
-    if (!is.null(by)) {
-      nodes <- lapply(sort(unique(edges[, by])), function(x) {
-        y <- nodes
-        y[, by] <- x
-        y
-      })
-      nodes <- do.call(rbind, nodes)
-    }
-
-    # return a data frame with network.size(model) + network.edgecount(model) rows,
-    # or length(unique(edges[, by ])) * network.size(model) + network.edgecount(model)
-    # rows if the nodes have been panelized
-
-    # [NOTE] `edges` has to be passed first to `rbind` in order for the edge
-    # attributes (e.g. factor) to be preserved in the result; this should not
-    # affect plotting the result, but differs from `ggnetwork` 0.5.1: `nodes`
-    # is now at the end of the result rather at the beginning
-    return(unique(rbind(edges[!is.na(edges$xend), ], nodes)))
-  } else {
-    # add missing columns to nodes data
-    nodes$xend <- nodes$x
-    nodes$yend <- nodes$y
-    return(nodes)
-  }
+  format_fortify(
+    model = model,
+    nodes = nodes,
+    weights = weights,
+    arrow.gap = arrow.gap,
+    by = by,
+    scale = scale,
+    stringsAsFactors = stringsAsFactors,
+    .lva = network::list.vertex.attributes,
```

Comment:
That's super smart, although I'll probably rename the functions to more explicit handles.

Reply:
I pushed a commit with "big names".

Reply:
Alright!